packages/playwright-core/src/server/page.ts

```diff
@@ -905,6 +906,22 @@ function addPageBinding(bindingName: string, needsHandle: boolean, utilityScript
   (globalThis as any)[bindingName].__installed = true;
 }
 
+export class InitScript {
+  readonly source: string;
+
+  constructor(source: string) {
+    const guid = createGuid();
+    this.source = `(() => {
+      globalThis.__pwInitScripts = globalThis.__pwInitScripts || {};
+      const hasInitScript = globalThis.__pwInitScripts[${JSON.stringify(guid)}];
```

Comment:
```suggestion
      const hasInitScript = globalThis.__pwInitScripts['${guid}'];
```
Here and below?

Reply:
I like JSON.stringify for consistency and safety when concating scripts.